### ISSUE TITLE
Issues with db.php and environment variables

### ISSUE THREAD

**Issue by baileydoestech** [nystudio107/craft-multi-environment#3]
We are seeing the following intermittent error (1 in 5 loads) in the admin dashboard using this approach for environment variables. 

Level	error
Category	system.db.CDbCommand
Message	CDbCommand::fetchAll() failed: SQLSTATE[3D000]: Invalid catalog name: 1046 No database selected. The SQL statement executed was: SHOW TABLES LIKE 'craft_%'.

Using literal values for username, password etc in db.php seems to be much more stable and the error is not seen.

**Comment by khalwat** [comment]
This is not an issue with CME, something is wrong with your environment.

**Comment by pl-mnm** [comment]
@lukebailey I'm having the exact same issue, switching back to crafts' `db.php` fixes it. Did you manage to find the source of the problem?

**Comment by khalwat** [comment]
So all the CME does is load a PHP file, which then sets the `ENV` variables. The only way it could conceivably fail is if the `.env.php` file is not present.

If you find something, I've love to hear it; but I think it's unlikely to be an issue with CME, and rather an environmental issue.

**Comment by pl-mnm** [comment]
So I tried to dig in a bit more with fresh installs of Craft and CME, tried switching between PHP 5.6 and 7.1, downloaded ampss and tried with that instead (I usually work with xampp), and still no luck. I also played around with vhosts settings thinking that perhaps the environment cannot access the `.env.php` for some reason, but my knowledge is these things is quite limited.

For what it's worth, with CME installed a whole bunch of CP JS files aren't loaded, and the CP sometimes doesn't load properly.

I have a feeling it's something really stupid that's staring me in the face and I can't see it. I'll give it another try later on.
